src/anyp/Uri.cc

```diff
@@ -933,8 +941,10 @@ AnyP::Uri::cleanup(const char *uri)
     char *cleanedUri = nullptr;
     switch (Config.uri_whitespace) {
     case URI_WHITESPACE_ALLOW:
-        flags |= RFC1738_ESCAPE_NOSPACE;
-    // fall through to next case
+        flags |= RFC1738_ESCAPE_NOSPACE | RFC1738_ESCAPE_UNESCAPED;
```

Comment:
AFAICT, this refactoring should also make `flags` a local-to-case constant:

```suggestion
        const auto flags = RFC1738_ESCAPE_NOSPACE | RFC1738_ESCAPE_UNESCAPED;
```

Same for other cases, of course.

Reply:
done

Reply:
actually it wasn't done. Now it is.